src/frequenz/sdk/timeseries/battery_pool/_metric_calculator.py

```diff
@@ -0,0 +1,478 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""Module that defines how to aggregate metrics from battery-inverter components."""
+import logging
+from abc import ABC, abstractmethod
+from datetime import datetime, timezone
+from typing import Dict, Generic, Iterable, List, Optional, Set, TypeVar
+
+from ... import microgrid
+from ...microgrid.component import ComponentCategory, ComponentMetricId
+from ...timeseries import Sample
+from ._component_metrics import ComponentMetricsData
+from ._result_types import Bound, CapacityMetrics, PowerBoundsMetrics, SoCMetrics
+
+_logger = logging.getLogger(__name__)
+_MIN_TIMESTAMP = datetime.min.replace(tzinfo=timezone.utc)
+
+
+def battery_inverter_mapping(batteries: Iterable[int]) -> Dict[int, int]:
```

Comment:
Super nitpick, but if this is only used in this module I would make it even *more private*.
```suggestion
def _battery_inverter_mapping(batteries: Iterable[int]) -> Dict[int, int]:
```

Or just move it as a private method of `PowerBoundsCalculator` to make the scope even more limited if it's just an utility function used only by that class.

Reply:
Done

Reply:
I can't see that, in the current state I see the name is still public but now it's being used somewhere else (`_methods.py`), so all good anyway. Feel free to resolved this one!